src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,140 @@
+	public static final double ANGLE_TOLERANCE = 0.5;
+
+	// HARDWARE
+	private final CANSparkFlex launcherTopMotor;
+	private final CANSparkFlex launcherBottomMotor;
+	private final CANSparkFlex launcherAngleMotor;
+	private final CANSparkFlex launcherHoodMotor;
+	private final RelativeEncoder launcherTopEncoder;
+	private final RelativeEncoder launcherBottomEncoder;
+	private final SparkAbsoluteEncoder launcherAngleEncoder;
+	private final SparkAbsoluteEncoder launcherHoodEncoder;
+	private final SparkPIDController launcherAnglePIDController;
+	private final SparkPIDController launcherHoodPIDController;
+
+	// digitalInput (sensor)
+	DigitalInput sensor1 = new DigitalInput(Hardware.LAUNCHER_SENSOR_1);
+
+	GenericEntry launcherSpeed =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher Speed", SPEAKER_SHOOT_SPEED)
+					.withSize(2, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+
+	GenericEntry launcherAngle =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher angle", getAngle())
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+
+	GenericEntry isNote =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("is loaded", false)
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kBooleanBox)
+					.getEntry();
+	// Constructor
+	public LauncherSubsystem() {
+
+		// MOTOR INSTANCE VARIBLES
+		launcherTopMotor = new CANSparkFlex(Hardware.LAUNCHER_TOP_MOTOR_ID, MotorType.kBrushless);
+		launcherBottomMotor = new CANSparkFlex(Hardware.LAUNCHER_BOTTOM_MOTOR_ID, MotorType.kBrushless);
+		launcherAngleMotor = new CANSparkFlex(Hardware.LAUNCHER_ANGLE_MOTOR_ID, MotorType.kBrushless);
+		launcherHoodMotor = new CANSparkFlex(Hardware.LAUNCHER_HOOD_MOTOR_ID, MotorType.kBrushless);
+		launcherTopEncoder = launcherTopMotor.getEncoder();
+		launcherBottomEncoder = launcherBottomMotor.getEncoder();
+		launcherAngleEncoder = launcherAngleMotor.getAbsoluteEncoder(Type.kDutyCycle);
+		launcherHoodEncoder =
+				launcherHoodMotor.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+		launcherAnglePIDController = launcherAngleMotor.getPIDController();
+		launcherHoodPIDController = launcherHoodMotor.getPIDController();
+
+		launcherAnglePIDController.setFeedbackDevice(launcherAngleEncoder);
+		launcherHoodPIDController.setFeedbackDevice(launcherHoodEncoder);
+		configMotors();
+	}
+
+	public void configMotors() {
+		launcherTopMotor.restoreFactoryDefaults();
+		launcherBottomMotor.restoreFactoryDefaults();
+		launcherAngleMotor.restoreFactoryDefaults();
+		launcherHoodMotor.restoreFactoryDefaults();
+
+		// idle mode (wow)
+		launcherTopMotor.setIdleMode(IdleMode.kBrake);
+		launcherBottomMotor.setIdleMode(IdleMode.kBrake);
+		launcherAngleMotor.setIdleMode(IdleMode.kBrake);
+		launcherHoodMotor.setIdleMode(IdleMode.kBrake);
+
+		// current limit
+		launcherTopMotor.setSmartCurrentLimit(20);
+		launcherBottomMotor.setSmartCurrentLimit(20);
```

Comment:
Likely should set the bottom motor to run inverted from the top motor, and also have it follow the top motor's settings so they run at the same velocity. Though disregard if there's a need to run the top & bottom motors at different velocities.

Reply:
There will be a need to run them at different velocities because of the way we want to score AMP unfortunately